src/widgets/tooltip.rs

```diff
@@ -0,0 +1,213 @@
+use std::sync::{Arc};
+use crate::core::{
+    Bound, Children, Color, EventType, MutableBound, OnEvent, rsx, widget,
+    render_command::RenderCommand, 
+    styles::{PositionType, Style, StyleProp, Units}
+};
+
+use crate::widgets::{Background, Clip, Element, If, Text};
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub struct TooltipData {
+    /// The anchor coordinates in pixels (x, y)
+    ///
+    /// If `None`, defaults to cursor position
+    pub anchor: Option<(f32, f32)>,
+    /// Used to force an update
+    ticker: u8,
```

Comment:
Does it not update when any of the other fields here change?

Reply:
It does. Sorry, this might actually not be needed anymore. I can remove it and the `mark_dirty` method.

I originally had to do this because I set the `anchor` directly on `TooltipProvider` instead of going through the `TooltipConsumer`. And we needed a way of telling `TooltipProvider` to update when the cursor moved (since the other fields didn't change).